### ISSUE TITLE
Rooms and Users should have a context menu in search results the same as if they’re shown in a room or member list

### ISSUE THREAD

**Issue by justjanne** [vector-im/element-web#22675]
The new search dialog needs to be accessible, especially for screen reader users.

Resources:

- https://github.com/matrix-org/element-web-rageshakes/issues/10871
- https://github.com/vector-im/element-web/issues/22361

> It might also be nice to right-click rooms and get the usual context menu. I guess this isn't what the search experience is strictly for, but if I just joined a room to troubleshoot something and want to leave it, being able to do so right from the search dialog would be convenient.

**Comment by justjanne** [comment]
This issue was split out from https://github.com/vector-im/element-web/issues/22361, copying the applicable labels from that issue.
